envs/uav_tracking.py

```diff
@@ -47,42 +52,75 @@ def __init__(self, discrete_actions = True,ep_lenght=50):
 
     
     def _get_obs(self):
-        pass
 
+        return {'agent': self.uav.position, 'target': self.uav.target_position}
+    
     def render(active= False):
         #Check how disable render ate unreal
         pass
 
-    def _compute_reward(self):
+    def _compute_reward(self,achieved_goal, desired_goal, info):
         """
         To develop a simple reward mechanic for the drone to go from point A to point B, 
-        We define a reward based on the distance traveled by the drone in the direction of the target,
-        calculating the distance between the current position of the drone and the target position, 
-        and then subtract the distance in the next time step from the current distance. 
-        This difference would give us the progress the drone made towards the target, and we can use this as the reward.
+        If the agent is within the distance_threshold of the target point, it receives a reward of 1.0. 
+        If it is farther away, the reward is calculated based on how far it is from the target point relative to the maximum distance the agent can be from the target
+        point (self.max_distance). This ensures that the reward is always between -1.0 and 0.0.
         """
+    # Check if the agent has reached the goal position
+        distance = np.linalg.norm(achieved_goal - desired_goal)
 
-        drone_pos = self._get_obs()["agent"]
-        target_pos = self._get_obs()["target"]
-        distance_to_target = np.linalg.norm(drone_pos - target_pos)
+        # Check if the agent has reached the goal position
+        if distance < self.threshold:
+            reward = 1.0
+        else:
+            # Compute a reward based on the distance to the goal
+            if distance < self.max_distance:
+                reward = (self.max_distance - distance) / self.max_distance
+            else:
+                reward = 0.0
 
-        # Move the drone in the direction of the target and measure the progress
-        self.uav.discrete_action(2)  # Move North
-        new_drone_pos = self._get_obs()["agent"]
-        new_distance_to_target = np.linalg.norm(new_drone_pos - target_pos)
-        distance_moved_towards_target = distance_to_target - new_distance_to_target
+        # Penalize the agent if it tries to move backwards
+        if info['action'] == 1 and achieved_goal[0] < self.current_position[0]:
```

Comment:
action == 1 significa subir o drone, além de quem o alvo pode estar em qualquer direção, logo penalizar por tomar uma ação especifica não se encaixa, a verificação deve ser se a distancia euclidiana do step atual for maior do que a do step anterior, significa que ele esta se afastando, e logo cabe penalidade

Reply:
Okk! Atualizei a função para incluir uma penalidade se o drone se afastar do objetivo em relação ao seu estado anterior. Além disso, como a ação "1" pode significar subir ou descer, removi a penalidade por tomar essa ação específica.